### PR TITLE
Add database support for addons

### PR DESCRIPTION
**Pull request by thani-sh**
Add a new addon database feature which can be used by addons to persist data with the storybook. The default configuration stores data in a JSON file inside the storybook config directory (.storybook).

### CODE REVIEW

File: src/client/manager/provider.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import React from 'react';
 import { Provider } from '@kadira/storybook-ui';
 import addons from '@kadira/storybook-addons';
 import createChannel from '@kadira/storybook-channel-pagebus';
+import createDatabase from '@kadira/storybook-database-local';
```
Comment (arunoda) [inline src/client/manager/provider.js]: Do we need to move this to another repo? If there's a reason we can move the server code too. (I mean the middleware)

Comment (mnmtanish) [inline src/client/manager/provider.js]: We need this on RN storybook also. I'll move the middle ware tomorrow. Please hold this PR until then.

Comment (arunoda) [inline src/client/manager/provider.js]: Move it to the same module. We can import the middleware like this:

~~~js
import middleware from '@kadira/storybook-database-local/dist/middleware'
~~~

**Comment by arunoda** [comment]
This is cool.

File: src/server/index.js [hunk 2]
```diff
@@ -64,7 +65,9 @@ if (program.staticDir) {
 // Build the webpack configuration using the `baseConfig`
 // custom `.babelrc` file and `webpack.config.js` files
 const configDir = program.configDir || './.storybook';
+const dbPath = path.resolve(configDir, 'datastore.json');
```
Comment (arunoda) [inline src/server/index.js]: For now, we should disable this by default.
We should enable saving data, if a user asked to do that via an option.

We can change that later, but until we figure things out this needs to be opt-in.

Comment (mnmtanish) [inline src/server/index.js]: How about `-d -- enable-db` flags for start-storybook command?

Comment (arunoda) [inline src/server/index.js]: Yep. That's good.

**Comment by arunoda** [comment]
I think we can get this. Do a release.

**Comment by mnmtanish** [comment]
Released `v2.7.0`

**Pull request merged**